src/connection/HttpConnection.ts

```diff
@@ -299,6 +298,7 @@ export default class HttpConnection extends BaseConnection {
         request.removeListener('response', onResponse)
         request.removeListener('timeout', onTimeout)
         request.removeListener('error', onError)
+        request.on('error', noop)
```

Comment:
The `.on` is intentional here (it's the main change of this PR). Not being subscribed to `error` while the stream is still ongoing is a risk of having unhandled `error` events that lead to the app crashing.

This listener is removed in the handler `onEnd` https://github.com/elastic/elastic-transport-js/blob/006e755a3a99ec4bc3f498e10a10719d3bf4a4ea/src/connection/HttpConnection.ts#L195, so I'd expect it to be cleaned up when the stream ends or is destroyed.

That being said, we could prob just attach the listener when the request is created and leave it there until the `onEnd` cleans it (instead of adding it in multiple places). WDYT?

Reply:
ahh thanks, I misread that. :man_facepalming: